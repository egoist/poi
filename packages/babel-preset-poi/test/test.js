const babel = require('babel-core')

function snapshot({ title, input }) {
  test(title, () => {
    const { code } = babel.transform(input, {
      presets: [require.resolve('..')]
    })
    expect(code).toMatchSnapshot(title)
  })
}

snapshot({
  title: 'object reset spread',
  input: 'const a = {...foo}'
})

snapshot({
  title: 'decorators legacy',
  input: `@bound class Foo {}`
})

snapshot({
  title: 'class properties',
  input: 'class Foo {state = 1}'
})
