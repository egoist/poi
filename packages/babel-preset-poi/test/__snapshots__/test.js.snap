// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class properties: class properties 1`] = `
"\\"use strict\\";

let Foo = class Foo {
  constructor() {
    this.state = 1;
  }

};"
`;

exports[`decorators legacy: decorators legacy 1`] = `
"\\"use strict\\";

var _class;

let Foo = bound(_class = class Foo {}) || _class;"
`;

exports[`object reset spread: object reset spread 1`] = `
"\\"use strict\\";

const a = Object.assign({}, foo);"
`;
